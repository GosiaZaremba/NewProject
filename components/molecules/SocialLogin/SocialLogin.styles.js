import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  socialContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    padding: 10,
  },
});
