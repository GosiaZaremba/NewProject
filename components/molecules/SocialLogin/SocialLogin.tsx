import React from 'react';
import {View} from 'react-native';
import {IconFB, IconGoogle} from '../../atoms';
import styles from './SocialLogin.styles';

export type Props = {
  onPressGoogle: () => void;
  onPressFacebook: () => void;
};

export const SocialLogin: React.FC<Props> = ({
  onPressFacebook,
  onPressGoogle,
}) => {
  return (
    <View style={styles.socialContainer}>
      <IconFB onPressFacebook={onPressFacebook}></IconFB>
      <IconGoogle onPressGoogle={onPressGoogle}></IconGoogle>
    </View>
  );
};
