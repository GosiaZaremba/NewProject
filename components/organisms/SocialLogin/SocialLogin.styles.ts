import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  socialContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginVertical: 10,
  },
});
