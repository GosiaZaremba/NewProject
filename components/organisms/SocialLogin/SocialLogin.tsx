import React from 'react';
import { View } from 'react-native';
import { Colors } from '../../../constants/colors';
import { CustomIcon } from '../../atoms';
import styles from './SocialLogin.styles';

export type Props = {
    onPressGoogle: () => void;
    onPressFacebook: () => void;
    onPressLinkedin: () => void;
};

export const SocialLogin: React.FC<Props> = ({
    onPressFacebook,
    onPressGoogle,
    onPressLinkedin,
}) => {
    return (
        <View style={styles.socialContainer}>
            <CustomIcon
                onPressIcon={onPressGoogle}
                iconColor={Colors.red}
                iconName={'google'}
                iconSize={20}
                iconBorderColor={Colors.red}
                testID={'googleIcon'}
                testOnly_pressed={true}
            />
            <CustomIcon
                onPressIcon={onPressFacebook}
                iconColor={Colors.navy}
                iconName={'facebook'}
                iconSize={20}
                iconBorderColor={Colors.navy}
                testID={'facebookIcon'}
                testOnly_pressed={true}
            />
            <CustomIcon
                onPressIcon={onPressLinkedin}
                iconColor={Colors.blue}
                iconName={'linkedin'}
                iconSize={20}
                iconBorderColor={Colors.blue}
                testID={'linkedinIcon'}
                testOnly_pressed={true}
            />
        </View>
    );
};
