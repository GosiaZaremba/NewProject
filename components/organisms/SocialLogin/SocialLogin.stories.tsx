/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { SocialLogin } from './SocialLogin';

storiesOf('organisms/SocialLogin', module)
    .addDecorator((story) => story())
    .add('default', () => (
        <SocialLogin
            onPressGoogle={() => {}}
            onPressFacebook={() => {}}
            onPressLinkedin={() => {}}
        />
    ));
