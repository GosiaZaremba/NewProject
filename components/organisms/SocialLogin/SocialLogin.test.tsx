import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import { SocialLogin } from './SocialLogin';

const mockOnPressGoogle = jest.fn();
const mockOnPressFacebook = jest.fn();
const mockOnPressLinked = jest.fn();

describe('Organism SocialLogin', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <SocialLogin
                    onPressGoogle={mockOnPressGoogle}
                    onPressFacebook={mockOnPressFacebook}
                    onPressLinkedin={mockOnPressLinked}
                    testID={'organism-socialLogin'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure google icon works correctly', () => {
        const { getByTestId } = render(
            <SocialLogin
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                testID={'organism-socialLogin'}
            />
        );

        const googleIcon = getByTestId('organism-socialLogin-google-icon');
        expect(googleIcon.props.style[1].fontSize).toEqual(20);
        fireEvent.press(googleIcon);
        expect(mockOnPressGoogle).toHaveBeenCalledTimes(1);
    });

    test('ensure facebook icon works correctly', () => {
        const { getByTestId } = render(
            <SocialLogin
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                testID={'organism-socialLogin'}
            />
        );

        const facebookIcon = getByTestId('organism-socialLogin-facebook-icon');
        expect(facebookIcon.props.style[1].fontSize).toEqual(20);
        fireEvent.press(facebookIcon);
        expect(mockOnPressFacebook).toHaveBeenCalledTimes(1);
    });

    test('ensure linkedIN icon works correctly', () => {
        const { getByTestId } = render(
            <SocialLogin
                onPressGoogle={mockOnPressGoogle}
                onPressFacebook={mockOnPressFacebook}
                onPressLinkedin={mockOnPressLinked}
                testID={'organism-socialLogin'}
            />
        );

        const linkedInIcon = getByTestId('organism-socialLogin-linkedIn-icon');
        expect(linkedInIcon.props.style[1].fontSize).toEqual(20);
        fireEvent.press(linkedInIcon);
        expect(mockOnPressLinked).toHaveBeenCalledTimes(1);
    });
});
