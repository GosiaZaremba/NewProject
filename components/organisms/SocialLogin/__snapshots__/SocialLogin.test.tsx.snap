// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism SocialLogin Component Snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 10,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 50,
          "borderWidth": 3,
          "height": 45,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 0.5,
          "padding": 10,
          "width": 45,
        },
        Object {
          "borderColor": "#e63039",
        },
      ]
    }
    testID="organism-socialLogin-google-pressable"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#e63039",
            "fontSize": 12,
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="organism-socialLogin-google-icon"
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 50,
          "borderWidth": 3,
          "height": 45,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 0.5,
          "padding": 10,
          "width": 45,
        },
        Object {
          "borderColor": "#39539b",
        },
      ]
    }
    testID="organism-socialLogin-facebook-pressable"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#39539b",
            "fontSize": 12,
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="organism-socialLogin-facebook-icon"
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 50,
          "borderWidth": 3,
          "height": 45,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 0.5,
          "padding": 10,
          "width": 45,
        },
        Object {
          "borderColor": "#3872c3",
        },
      ]
    }
    testID="organism-socialLogin-linkedIn-pressable"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#3872c3",
            "fontSize": 12,
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="organism-socialLogin-linkedIn-icon"
    >
      
    </Text>
  </View>
</View>
`;
